hail/src/main/scala/is/hail/shuffler/Wire.scala

```diff
@@ -0,0 +1,70 @@
+package is.hail.shuffler
+
+import is.hail.HailLSM
+import is.hail.annotations.Region
+import is.hail.expr.ir.IRParser
+import is.hail.expr.types.encoded.EType
+import is.hail.expr.types.physical.{ PStruct, PType }
+import is.hail.expr.types.virtual.{ TStruct, Type }
+import is.hail.io.TypedCodecSpec
+import is.hail.rvd.AbstractRVDSpec
+import java.io._
+import java.net._
+import java.security.KeyStore;
+import java.util.UUID
+import java.util.concurrent.{ ConcurrentSkipListMap, Executors }
+import javax.net._
+import javax.net.ssl._
+import javax.security.cert.X509Certificate;
+import org.json4s.jackson.{ JsonMethods, Serialization }
+import com.fasterxml.jackson.core.{ JsonGenerator, JsonParser }
+
+import scala.annotation.switch
+
+import is.hail.utils._
+
+object Wire {
+  val START: Byte = 0.toByte
+  val PUT: Byte = 1.toByte
+  val GET: Byte = 2.toByte
+
+  import AbstractRVDSpec.formats
+
+  // Jackson closes the InputStreams and OutputStreams you pass to the
+  // ObjectMapper unless you set these flags, respectively, to false
+  JsonMethods.mapper.configure(JsonParser.Feature.AUTO_CLOSE_SOURCE, false);
+  JsonMethods.mapper.configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false);
```

Comment:
I'm increasingly against the extract business.  If you want `fromJValue` and `toJValue`, I'd be happy.  The other option is to pull out a utf-8 string and then extract from that, instead of sending in your socket stream.

Reply:
Agreed. If provided with fromJValue and toJValue I would happily use them in the shuffler. Until then, I'm fine with wrapping up annoying IO logic in this class.